Sources/Config/ConfigurationProperty.swift

```diff
@@ -26,21 +29,39 @@ struct ConfigurationProperty<T>: Property, AssociatedPropertyKeyProviding {
         }
     }
 
+    private static func transformValueToType(value: Any?) throws -> T {
+        if let val = value as? T {
+            return val
+        }
+        if let nilLiteralType = T.self as? ExpressibleByNilLiteral.Type {
+            return nilLiteralType.init(nilLiteral: ()) as! T
+        }
+
+        throw Failure.notConvertible
+    }
+
     init?(key: String, typeHint: String, dict: [String: Any]) {
-        guard let defaultValue = dict["defaultValue"] as? T else {
+        do {
+            self.defaultValue = try ConfigurationProperty.transformValueToType(value: dict["defaultValue"])
+            
+            self.key = key
+            self.typeHint = typeHint
+            self.associatedProperty = dict["associatedProperty"] as? String
+            self.type = PropertyType(rawValue: typeHint)
+            self.description = dict["description"] as? String
+
+            let overrides = try? dict["overrides"]
+                .flatMap { $0 as? [String: Any] }?
```

Comment:
Should this be `compactMap`?

Reply:
In this case we're actually `flatMap`'ing the `Optional` rather than the `Collection`

Reply:
Fair enough